From f383fa4b9c64e319cb9b8a63ca88368ad752db27 Mon Sep 17 00:00:00 2001
From: Albert Astals Cid <aacid@kde.org>
Date: Wed, 30 Jan 2019 01:31:30 +0100
Subject: [PATCH 098/166] png: initialize color_type to 0

Fixes
==12==WARNING: MemorySanitizer: use-of-uninitialized-value
    #0 0x6b8179 in setup_qt /src/qtbase/src/gui/image/qpnghandler.cpp:247:9
on fuzzed file

Change-Id: I772d536a0db91665dc16e94751ef507de1064376
Reviewed-by: Eirik Aavitsland <eirik.aavitsland@qt.io>
---
 src/gui/image/qpnghandler.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/gui/image/qpnghandler.cpp b/src/gui/image/qpnghandler.cpp
index 8ae03d5d38..110ee1670b 100644
--- a/src/gui/image/qpnghandler.cpp
+++ b/src/gui/image/qpnghandler.cpp
@@ -234,7 +234,7 @@ void setup_qt(QImage& image, png_structp png_ptr, png_infop info_ptr, QSize scal
     png_uint_32 width;
     png_uint_32 height;
     int bit_depth;
-    int color_type;
+    int color_type = 0;
     png_bytep trans_alpha = 0;
     png_color_16p trans_color_p = 0;
     int num_trans;
@@ -678,7 +678,7 @@ QImage::Format QPngHandlerPrivate::readImageFormat()
 {
         QImage::Format format = QImage::Format_Invalid;
         png_uint_32 width, height;
-        int bit_depth, color_type;
+        int bit_depth, color_type = 0;
         png_colorp palette;
         int num_palette;
         png_get_IHDR(png_ptr, info_ptr, &width, &height, &bit_depth, &color_type, 0, 0, 0);
-- 
2.20.1

